docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,271 @@
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/design/aws_under_the_hood.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Peeking under the hood of kubernetes on AWS
+
+We encourage you to use kube-up (or CloudFormation) to create a cluster.  But
+it is useful to know what is being created: for curiosity, to understand any
+problems that may arise, or if you have to create things manually because the
+scripts are unsuitable for any reason.  We don't recommend manual configuration
+(please file an issue and let us know what's missing if there's something you
+need) but sometimes it is the only option.
+
+This document sets out to document how kubernetes on AWS maps to AWS objects.
+Familiarity with AWS is assumed.
+
+### Top-level
+
+Kubernetes consists of a single master node, and a collection of minion nodes.
+Other documents describe the general architecture of Kubernetes (all nodes run
+Docker; the kubelet agent runs on each node and launches containers; the
+kube-proxy relays traffic between the nodes etc).
+
+By default on AWS:
+
+* Instances run Ubuntu 15.04 (the official AMI).  It includes a sufficiently
+  modern kernel to give a good experience with Docker, it doesn't require a
+  reboot.  (The default SSH user is `ubuntu` for this and other ubuntu images)
+* By default we run aufs over ext4 as the filesystem / container storage on the
+  nodes (mostly because this is what GCE uses).
+
+These defaults can be changed by passing different environment variables to
+kube-up.
+
+### Storage
+
+AWS does support persistent volumes via EBS.  These can then be attached to
+pods that should store persistent data (e.g. if you're running a database).
+
+Minions do not have persistent volumes otherwise.  In general, kubernetes
+containers do not have persistent storage unless you attach a persistent
+volume, and so minions on AWS use instance storage.  Instance storage is
+cheaper, often faster, and historically more reliable.  This does mean that you
+should pick an instance type that has sufficient instance storage, unless you
```

Comment:
Although everyone uses instance storage: our containers are stored on instance storage.  If you are running e.g. mostly webservers you can probably make do with not a lot of disk because your is image is probably pretty small and you're not going to be creating a lot of data on disks.  But if either of those aren't true, you have to choose an instance type that includes more instance storage.

Right now the big gotcha here is that elasticsearch & influxdb create their data on instance storage.  They shouldn't IMHO, but they do...